### PR TITLE
feat: add rand and trunc expressions

### PR DESCRIPTION
**Pull request by yvonnep165**
Adds support for the arithmetic pipeline expressions `rand` and `trunc`.
Since a RandomExpression class already exists internally, I make it public and add a global rand() function there. 

ported from https://github.com/firebase/firebase-js-sdk/pull/9498

### CODE REVIEW

File: Firestore/Swift/Source/SwiftAPI/Pipeline/Expressions/Expression.swift [hunk 1]
```diff
@@ -139,6 +139,45 @@ public protocol Expression: Sendable {
...
+  /// // Get the value of the "rating" field truncated to an integer.
+  /// Field("rating").trunc()
+  /// ```
+  ///
+  /// - Returns: A new `FunctionExpression` representing the truncated number.
+  func trunc() -> FunctionExpression
+
+  /// Creates an expression that truncates self to a specified number of decimal places.
+  ///
+  /// If `decimalPlace` is positive, truncates names to the right of the decimal point.
```
Comment (milaGGL) [inline Firestore/Swift/Source/SwiftAPI/Pipeline/Expressions/Expression.swift]: a bit confused about this part. how "names" are truncated? how negative decimal place works? like "1.2345" => "1.234" if `truncToPrecision(-1)`?

File: Firestore/Swift/Tests/Integration/PipelineTests.swift [hunk 2]
```diff
@@ -2351,6 +2351,76 @@ class PipelineIntegrationTests: FSTIntegrationTestCase {
...
+    try await collRef.document("dummy").setData(["a": 1])
+
+    let pipeline = db.pipeline()
+      .collection(collRef.path)
+      .limit(1)
+      .select([
+        Constant(4.123456).truncToPrecision(0).as("p0"),
+        Constant(4.123456).truncToPrecision(1).as("p1"),
+        Constant(4.123456).truncToPrecision(Constant(2)).as("p2"),
+        Constant(4.123456).truncToPrecision(4).as("p4"),
```
Comment (milaGGL) [inline Firestore/Swift/Tests/Integration/PipelineTests.swift:2458]: add negative number test case

File: Firestore/Swift/Tests/Integration/PipelineTests.swift [hunk 3]
```diff
@@ -2351,6 +2351,76 @@ class PipelineIntegrationTests: FSTIntegrationTestCase {
...
+      .pipeline()
+      .collection(collRef.path)
+      .select([rand().as("randomNumber")])
+      .limit(1)
+      .execute()
+
+    XCTAssertEqual(snapshot.results.count, 1)
+    let data = snapshot.results[0].data
+    guard let randomNumber = data["randomNumber"] as? Double else {
+      XCTFail("randomNumber is not a Double")
```
Comment (milaGGL) [inline Firestore/Swift/Tests/Integration/PipelineTests.swift:2494]: not a double, so float?

File: Firestore/Swift/Source/SwiftAPI/Pipeline/Expressions/Expression.swift (hunk 1 above)
Comment (yvonnep165) [reply Firestore/Swift/Source/SwiftAPI/Pipeline/Expressions/Expression.swift]: Sorry for the typo. Should be "number".For negative decimal place, truncToPrecision(-1) for 1.2345 will be 0.0,  truncToPrecision(-1) for 12.3456 will be 10.0. Corrected!

File: Firestore/Swift/Tests/Integration/PipelineTests.swift (hunk 3 above)
Comment (yvonnep165) [reply Firestore/Swift/Tests/Integration/PipelineTests.swift:2494]: This is to confirm it is the "double" data type. If not it will raise an error

File: Firestore/Swift/Tests/Integration/PipelineTests.swift (hunk 2 above)
Comment (yvonnep165) [reply Firestore/Swift/Tests/Integration/PipelineTests.swift:2458]: Negative number test cases added. But I'm a bit confused about the scope of the tests to be covered. Do we need to include every possible case in the integration test? Asking since I think we will rely on the backend's tests to cover all the cases

File: Firestore/Swift/Tests/Integration/PipelineTests.swift (hunk 3 above)
Comment (milaGGL) [reply Firestore/Swift/Tests/Integration/PipelineTests.swift:2494]: ah, i see.

File: Firestore/Swift/Tests/Integration/PipelineTests.swift (hunk 2 above)
Comment (milaGGL) [reply Firestore/Swift/Tests/Integration/PipelineTests.swift:2458]: not required to cover all cases , but nice to have this test cases to illustrate how "If `decimalPlace` is negative, truncates digits to the left of the decimal point." works.
